### PR TITLE
[mtl] add missing limits

### PR DESCRIPTION
**Pull request by aleksijuvani**
Fixes #issue
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:
- [ ] `rustfmt` run on changed code

### CODE REVIEW

File: src/backend/metal/src/device.rs [hunk 1]
```diff
@@ -449,8 +449,13 @@ impl hal::PhysicalDevice<Backend> for PhysicalDevice {
             max_push_constants_size: 0x1000,
             max_sampler_allocation_count: !0,
             max_bound_descriptor_sets: MAX_BOUND_DESCRIPTOR_SETS as _,
-            max_descriptor_set_uniform_buffers: 0x100, // also arbitrary, but probably shouldn't be more than max_bound_descriptor_sets
-            max_fragment_input_components: 32, // TODO: Add to private caps since 32 is only for mac
+            max_descriptor_set_samplers: pc.max_samplers_per_stage as usize * 4,
```
Comment (kvark) [inline src/backend/metal/src/device.rs]: Where does "* 4" come from?

Comment (aleksijuvani) [reply src/backend/metal/src/device.rs]: This is the same logic as in MoltenVK, which seemed to make sense to me—if I'm not mistaken, there are four programmable shader stages, and if each supports a maximum of `max_samplers_per_stage`, the total ought to be `max_samplers_per_stage * 4`. Actually, looking at the [Metal Tessellation Pipeline](https://developer.apple.com/library/archive/documentation/Miscellaneous/Conceptual/MetalProgrammingGuide/Tessellation/Tessellation.html), maybe this ought to be three instead?

Comment (kvark) [reply src/backend/metal/src/device.rs]: Ok, I see. This makes sense.
Let's have a constant somewhere, i.e. `SHADER_STAGE_COUNT` to use for multiplication here. Its value should be `3`.

Comment (aleksijuvani) [reply src/backend/metal/src/device.rs]: Done!

**Pull request merged**